mock-services/DAG-parser.py

```diff
@@ -0,0 +1,337 @@
+        for s in services:
+            add_configmap += f"kubectl create configmap {s}-configmap --from-file {s}/config.yaml\n"
+
+        self.run_template = f"""#!/bin/bash
+{add_configmap}
+kubectl apply -f test.yaml
+
+sleep 10
+kubectl get svc
+echo
+serviceip=$(kubectl get svc | grep {first_service} | cut -d ' ' -f 9)
+kubectl get pods
+podname=$(kubectl get pods | grep {first_service} | cut -d ' ' -f 1)
+curl ${{serviceip}}:9080/saga-add/1
+sleep 1
+echo
+kubectl logs ${{podname}} qbox
+echo
+echo
+kubectl logs ${{podname}} {first_service}
+"""
+        delete_configmap = ""
+        for s in services:
+            delete_configmap += f"kubectl delete configmap {s}-configmap\n"
+        self.stop_template = f"""#!/bin/bash
+{delete_configmap}
+kubectl delete -f test.yaml        
+"""
+    def get_stop_script(self):
+        return self.stop_template
+
+    def get_run_script(self):
+        return self.run_template
+
+
+def config_generator(file, dir = '.'):
+    if os.path.isdir(os.path.join(dir, "test")):
+        print("here")
+        shutil.rmtree(os.path.join(dir, "test"))
+    # os.mkdir(os.path.join(dir, "test"))
+    shutil.copytree(os.path.join(dir, "lib"), os.path.join(dir, "test"))
+
+    f = open(file, "r")
+    obj = json.load(f)
+    f.close()
+    root = obj["root"]
+
+    test_yaml = []
+    services = []
+
+    for service in obj["nodes"]:
+        if service in obj["edges"]:
+            services.append(service)
+            os.mkdir(os.path.join(dir, "test", service))
+            f = open(os.path.join(dir, "test", service, "config.yaml"), "w")
+            file_template = FileTemplate()
+
+            file_template.construct_add(service, obj)
+            if root != service:
+                file_template_delete = FileTemplate()
+                file_template_delete.construct_delete(service, obj)
+                yaml.safe_dump_all(
+                    [file_template.template, file_template_delete.template], f)
+            else:
+                yaml.safe_dump(file_template.template, f)
+            f.close()
+        test_yaml.append(DeploymentTemplate(service, service in obj["bad"], service in obj["edges"]).get_file())
+    
+    for service in obj["nodes"]:
+        test_yaml.append(ServiceTemplate(service).get_file())
+    f = open(os.path.join(dir, "test", "test.yaml"), "w")
+    yaml.safe_dump_all(test_yaml, f)
+    f.close()
+    f = open(os.path.join(dir, "test", "run.sh"), "w")
+    t = ScriptTemplate(services, obj["root"])
+    f.write(t.get_run_script())
+    f.close()
+    f = open(os.path.join(dir, "test", "stop.sh"), "w")
+    f.write(t.get_stop_script())
+    f.close()
+    st = os.stat(os.path.join(dir, "test", "run.sh"))
+    os.chmod(os.path.join(dir, "test", "run.sh"), st.st_mode | stat.S_IEXEC)
+    st = os.stat(os.path.join(dir, "test", "stop.sh"))
+    os.chmod(os.path.join(dir, "test", "stop.sh"), st.st_mode | stat.S_IEXEC)
+
+    f = open(os.path.join(dir, "test", "values.tfvars"), "w")
```

Comment:
Rather than hardcode the file contents completely, can you parse the file as `=`-delimited lines, find the `add_configmap` key, and update it before writing to the file? This prevents sensitive values and machine-specific values being written.